mcs/class/System.Net.Http/System.Net.Http/HttpClient.android.cs

```diff
@@ -8,21 +8,25 @@ public partial class HttpClient {
 
 		static HttpMessageHandler CreateDefaultHandler ()
 		{
-			Type type = Type.GetType("Android.Runtime.AndroidEnvironment, Mono.Android");
-			if (type == null)
-				return GetFallback ("Invalid Mono.Android assembly? Cannot find Android.Runtime.AndroidEnvironment");
+			string envvar = Environment.GetEnvironmentVariable ("XA_HTTP_CLIENT_HANDLER_TYPE")?.Trim ();
+			if (envvar == "System.Net.Http.MonoWebRequestHandler")
+				return new HttpMessageHandler (new MonoWebRequestHandler ());
 
-			MethodInfo method = type.GetMethod ("GetHttpMessageHandler", BindingFlags.Static | BindingFlags.NonPublic);
```

Comment:
So I basically moved [AndroidEnvironment.GetHttpMessageHandler](https://github.com/xamarin/xamarin-android/blob/master/src/Mono.Android/Android.Runtime/AndroidEnvironment.cs#L358-L370) logic to the mono side
so we actually can delete that method in android (Linker will be fine).